src/backend/providers/playnite/PlayniteMetadataParser.cpp

```diff
@@ -0,0 +1,173 @@
+#include <QJsonDocument>
+#include <QJsonObject>
+
+namespace providers {
+namespace playnite {
+
+PlayniteMetadataParser::PlayniteMetadataParser(QString log_tag, const QDir& playnite_dir)
+    : m_log_tag(std::move(log_tag))
+    , m_playnite_dir(playnite_dir)
+    , dir_filters(QDir::Files | QDir::Readable | QDir::NoDotAndDotDot)
+    , json_ext_list(QStringList() << QStringLiteral("*.json"))
+{}
+
+PlayniteComponents PlayniteMetadataParser::parse_metdata() const
+{
+    PlayniteComponents components;
+    components.platforms = parse_platform_metadata();
+    components.sources = parse_source_metadata();
+    components.companies = parse_company_metadata();
+    components.genres = parse_genre_metadata();
+    components.emulators = parse_emulator_metadata();
+    components.games = parse_game_metadata();
+    return components;
+}
+
+HashMap<QString, QString> PlayniteMetadataParser::parse_platform_metadata() const
+{
+    return parse_id_name_files(QStringLiteral("library/platforms"));
+}
+
+HashMap<QString, QString> PlayniteMetadataParser::parse_source_metadata() const
+{
+    return parse_id_name_files(QStringLiteral("library/sources"));
+}
+
+HashMap<QString, QString> PlayniteMetadataParser::parse_company_metadata() const
+{
+    return parse_id_name_files(QStringLiteral("library/companies"));
+}
+
+HashMap<QString, QString> PlayniteMetadataParser::parse_genre_metadata() const
+{
+    return parse_id_name_files(QStringLiteral("library/genres"));
+}
+
+HashMap<QString, PlayniteEmulator> PlayniteMetadataParser::parse_emulator_metadata() const
+{
+    HashMap<QString, PlayniteEmulator> emulator_map;
+    QDirIterator dir_it(m_playnite_dir.filePath(QStringLiteral("library/emulators/")), json_ext_list, dir_filters);
+    while (dir_it.hasNext()) {
+        PlayniteEmulator emu;
+        const QString file_path = dir_it.next();
+        const QJsonObject json_object = get_json_object_from_file(file_path);
+        const auto emu_json = JsonObjectHelper(json_object);
+        emu.id = emu_json.get_string(QStringLiteral("Id"));
+        emu.name = emu_json.get_string(QStringLiteral("name"));
+        auto profiles = emu_json.get_json_array(QStringLiteral("Profiles"));
+        for (const auto profile : profiles) {
+            PlayniteEmulator::EmulatorProfile emu_profile;
+            auto profile_json = JsonObjectHelper(profile.toObject());
+            emu_profile.platform_ids = profile_json.get_string_list(QStringLiteral("Platforms"));
+            emu_profile.executable = profile_json.get_string(QStringLiteral("Executable"));
+            emu_profile.arguments = profile_json.get_string(QStringLiteral("Arguments"));
+            emu_profile.working_dir = profile_json.get_string(QStringLiteral("WorkingDirectory"));
+            emu_profile.id = profile_json.get_string(QStringLiteral("Id"));
+            emu_profile.name = profile_json.get_string(QStringLiteral("Name"));
+            emu.profiles.insert({ emu_profile.id, emu_profile });
```

Comment:
`emplace` (and `emplace_back` for vectors) is preferred over `insert`, for optimization.